com.unity.cinemachine/Runtime/Behaviours/CinemachineConfiner2D.cs

```diff
@@ -498,11 +511,70 @@ void CalculateDeltaTransformationMatrix()
+        internal bool IsCameraTooBigForTheConfiner(CinemachineVirtualCameraBase vcam)
+        {
+            InvalidateLensCacheIfNeeded(vcam);
+            
+            if (BoundingShape2D == null)
+                return false;
+            
+            if (m_ShapeCache.ConfinerOven != null && m_ShapeCache.ConfinerOven.m_Skeleton.Count > 0)
+                return true; // there is a skeleton, that means some parts are collapsed -> oversized
+            
+            if (vcam is CinemachineCameraManagerBase manager)
+            {
+                foreach (var childCamera in manager.ChildCameras)
+                    if (CheckForOversized(GetExtraState<VcamExtraState>(childCamera), m_ShapeCache))
+                        return true; // at least one child is oversized
+            }
+            else if (CheckForOversized(GetExtraState<VcamExtraState>(vcam), m_ShapeCache))
+                return true; // oversized
+            
+            return false;
+            
+            // local function
+            static bool CheckForOversized(VcamExtraState extra, ShapeCache shapeCache)
+            {
+                if (extra.BakedSolution != null)
+                {
+                    var solution = extra.BakedSolution.m_Solution;
+                    if (solution.Count == 1 && solution[0].Count == 1)
+                        return true; // shrank down to mid point -> oversized
+                    if (shapeCache.OriginalPath != null && solution.Count != shapeCache.OriginalPath.Count)
+                        return true; // polygon count of the input and solution differs -> oversized
+                }
+                return false;
+            }
+        }
+        
+        void InvalidateLensCacheIfNeeded(CinemachineVirtualCameraBase vcam)
+        {
+            if (vcam is CinemachineCameraManagerBase manager)
+            {
+                foreach (var childCamera in manager.ChildCameras)
+                    if (IsLensCacheInvalid(childCamera.State, GetExtraState<VcamExtraState>(childCamera), m_ShapeCache))
+                        InvalidateLensCache(childCamera);
+            }
+            else if (IsLensCacheInvalid(vcam.State, GetExtraState<VcamExtraState>(vcam), m_ShapeCache))
+                InvalidateLensCache(vcam);
+            
+            // local function
+            static bool IsLensCacheInvalid(CameraState state, VcamExtraState extra, ShapeCache shapeCache)
+            {
+                var lens = state.Lens;
+                var deltaW = shapeCache.DeltaWorldToBaked;
+                var frustum = 
+                    CalculateHalfFrustumHeight(lens, deltaW.MultiplyPoint3x4(state.GetCorrectedPosition()).z);
+                return Mathf.Abs(extra.AspectRatio - lens.Aspect) > UnityVectorExtensions.Epsilon ||
+                    Mathf.Abs(extra.FrustumHeight - frustum * deltaW.lossyScale.x) > UnityVectorExtensions.Epsilon;
+            }
+        }
```

Comment:
Should this be public?

If so, should it follow the pattern of InvalidateLensCache (that it ignores ManagerCameras)?
Or should InvalidateLensCache follow this pattern, where we check whether we are dealing with manager cameras.

What do you think?

Reply:
https://github.com/Unity-Technologies/com.unity.cinemachine/pull/759 resolved question